### PR TITLE
Support for React Native 0.75

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

- bump FabricExample to RN 0.75
- make changes necessary to build the app
  - native state type got changed from `ShadowNode` to `ShadowNodeWrapper`
- update `ghQueueMicrotask` to include `setImmediate` as `requestAnimationFrame` [is equivalent to `setTimeout(0)`  ](https://github.com/facebook/react-native/blob/1c1c8335db2494672cf955cf4db574e23fd2198a/packages/react-native/ReactCommon/react/runtime/TimerManager.cpp#L441-L443)on RN and seems not to be consistent in regards to the order of execution

## Test plan

Test on FabricExample and Example apps

### CODE REVIEW

File: android/src/main/jni/cpp-adapter.cpp [hunk 1]
```diff
@@ -18,11 +18,17 @@ void decorateRuntime(jsi::Runtime &runtime) {
...
+                auto arg = arguments[0].asObject(runtime);
+
+                // TODO: this condition always fails?
+                if (arg.hasNativeState<ShadowNodeWrapper>(runtime)) {
+                  auto shadowNodeWrapper = arg.getNativeState<ShadowNodeWrapper>(runtime);
+                  bool isFormsStackingContext = shadowNodeWrapper->shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
+                  return jsi::Value(isFormsStackingContext);
+                }
+
+                return jsi::Value(true);
```
Comment (tomekzaw) [inline android/src/main/jni/cpp-adapter.cpp:31]: Copied from Reanimated's fabric-example (it works)
```suggestion
                auto shadowNodeWrapper = arguments[0]
                        .asObject(runtime).getNativeState<ShadowNodeWrapper>(runtime);
                bool isFormsStackingContext = shadowNodeWrapper->shadowNode->getTraits()
                        .check(ShadowNodeTraits::FormsStackingContext);

                return jsi::Value(isFormsStackingContext);
```

**Comment by raghav-reglobe** [comment]
Hi, when can we expect this to be merged.

File: apple/RNGestureHandlerModule.mm [hunk 2]
```diff
@@ -99,10 +99,17 @@ void decorateRuntime(jsi::Runtime &runtime)
...
+        auto arg = arguments[0].asObject(runtime);
+
+        if (arg.hasNativeState<ShadowNodeWrapper>(runtime)) {
+          auto shadowNodeWrapper = arg.getNativeState<ShadowNodeWrapper>(runtime);
+          bool isFormsStackingContext =
+              shadowNodeWrapper->shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
+          return jsi::Value(isFormsStackingContext);
+        }
+
+        return jsi::Value(true);
```
Comment (tomekzaw) [inline apple/RNGestureHandlerModule.mm:112]: Can we unify the implementation here with `cpp-adapter.cpp`?

File: src/ghQueueMicrotask.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 // We check for typeof requestAnimationFrame because of SSR
+// Functions are bound to null to avoid issues with scope when using Metro inline requires.
 export const ghQueueMicrotask =
-  typeof requestAnimationFrame === 'function'
-    ? // Functions are bound to null to avoid issues with scope when using Metro inline requires.
-      requestAnimationFrame.bind(null)
+  typeof setImmediate === 'function'
+    ? setImmediate.bind(null)
```
Comment (tomekzaw) [inline src/ghQueueMicrotask.ts:5]: Why do we need this?

Comment (j-piasecki) [reply src/ghQueueMicrotask.ts:5]: It's in the PR description

> update ghQueueMicrotask to include setImmediate as requestAnimationFrame [is equivalent to setTimeout(0) ](https://github.com/facebook/react-native/blob/1c1c8335db2494672cf955cf4db574e23fd2198a/packages/react-native/ReactCommon/react/runtime/TimerManager.cpp#L441-L443)on RN and seems not to be consistent in regards to the order of execution

File: apple/RNGestureHandlerModule.mm (hunk 2 above)
Comment (j-piasecki) [reply apple/RNGestureHandlerModule.mm]: Done in [2e06198](https://github.com/software-mansion/react-native-gesture-handler/pull/2966/commits/2e0619802bb3db99a79f2ee7252b8374032b373c). Thanks for catching this!

**Pull request merged**